engine/runtime-parser/src/main/java/org/enso/compiler/core/TreeToIr.java

```diff
@@ -670,6 +670,16 @@ && isDotOperator(oprApp.getOpr().getRight())
               var loc = getIdentifiedLocation(oprApp.getLhs());
               args.add(new CallArgument.Specified(Option.empty(), self, loc, meta(), diag()));
             }
+          } else if (tree instanceof Tree.OprApp oprApp
+                  && isDotDotOperator(oprApp.getOpr().getRight())
```

Comment:
I've been just chatting with @4e6 about making sure the _autoscoped constructors_ with `..` prefix are properly handled by the IDE. We concluded that the IDE needs to do something similar to this check to recognize `..` as _autoscoped prefix_ rather than an **operator missing lhs** (which is the current behavior).

Dmitry suggested to treat `..` as _unary operator_ and have a `Tree` in our Rust parser for it. What do you think about it @kazcw? Should `..` be recognized in the Rust `Tree` or do we want special code like this one also in the IDE?

How do you want to support _autoscoped constructors_ with `..` prefix in the IDE, @farmaazon?

Reply:
I strongly prefer having parser interpret `..A` as unary operator.

Reply:
> I should have reviewed this PR.

You had been added as a reviewer manually (as you are not code owner) exactly for this reason.

> I presume we will not be supporting [the use of .. found in the spec](https://github.com/enso-org/design/blob/wip/wd/enso-spec/epics/enso-spec-1.0/02.%20Important%20concepts.md#motivation)?

That's question for @wdanilo. So far engine continues to report code like `1 .. 2` as _"no method .. found on Integer"_.

That said, I have to add: I don't know a syntax to add `..` method on any type.

Reply:
That's correct. The spec was very limited in this scope, and we can safely interpret `..` as autoscope operator. Regarding spec being limited - you want to be able to express `rangeFromTo`, `rangeFrom`, `rangeTo`, rangeToIncluding`, and `rangeFromToIncluding` using syntax. Using `..` operator would not allow for that, or would create a very strange corner cases regarding currying. If we want to support range operators, we can borrow the syntax from Haskell: `[1..]`, etc, where `..` is part of the array expression. But that has to be properly designed in the future.

Reply:
> I strongly prefer having parser interpret `..A` as unary operator.

- being implemented by Kaz as #9372